examples/flutter_gallery/lib/demo/cupertino/cupertino_navigation_demo.dart

```diff
@@ -0,0 +1,740 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+class CupertinoNavigationDemo extends StatelessWidget {
+  static const String routeName = '/cupertino/navigation';
+
+  @override
+  Widget build(BuildContext context) {
+    return new WillPopScope(
+      // Prevent swipe popping of this page. Use explicit exit buttons only.
+      onWillPop: () => new Future<bool>.value(true),
```

Comment:
don't we want to use fullScreenDialog for this?

Reply:
I think it would be nice. Though we'd first need some form of #6225 so pushNamed can specify it.